fix(useFetch): Применить рекомендации.

SuvStreet opened the pull request.

SuvStreet committed:
fix(useFetch): Применить рекомендации.

dmitry-rozhko commented on src/hooks/useFetch.js:
```diff
@@ -14,30 +14,29 @@ export function useFetch(url) {
+    async ({ params } = '') => {
+      try {
+        const response = await fetch(
+          params ? `${url}?_limit=${params._limit}` : url
```
жестко закодированный `_limit`  не думаю что это хорошая идея, подумай как это можно улучшить

dmitry-rozhko commented on src/hooks/useFetch.js:
```diff
@@ -14,30 +14,29 @@ export function useFetch(url) {
-    setIsLoading(true)
-    fetchData(params)
-  }
+  const fetchData = useCallback(
```
fetchData должен быть выше useEffect() так как у стрелочных функций уже нет эффекта всплытия

dmitry-rozhko commented on src/hooks/useFetch.js:
```diff
@@ -14,30 +14,29 @@ export function useFetch(url) {
     }
   }, [])
```
мы добавим зависимость fetchData

`}, [fetchData])`

это нужно для того, чтобы хук понимал, когда необходимо снова запустить запрос данных, например если изменился url

SuvStreet committed:
fix(useFetch): Исправить недочёты

1. Убрать жёстко заданный _limit (параметры к url)
2. Добавить к useEffect() зависимости options, fetchData
